Sources/GRPCInProcessTransport/InProcessServerTransport.swift

```diff
@@ -55,12 +55,17 @@ public struct InProcessServerTransport: ServerTransport, Sendable {
     }
   }
 
-  /// Return a new ``RPCAsyncSequence`` that will contain all published ``RPCStream``s published
-  /// to this transport using the ``acceptStream(_:)`` method.
-  ///
-  /// - Returns: An ``RPCAsyncSequence`` of all published ``RPCStream``s.
-  public func listen() async throws -> RPCAsyncSequence<RPCStream<Inbound, Outbound>> {
-    RPCAsyncSequence(wrapping: self.newStreams)
+  public func listen(
+    _ streamHandler: @escaping (RPCStream<Inbound, Outbound>) async throws -> Void
+  ) async throws {
+    try await withThrowingTaskGroup(of: Void.self) { group in
```

Comment:
Can we use a discarding task group here? Also we probably shouldn't fail the whole transport when we fail to handle a single stream

Reply:
Actually, this being throwing was left over from one of the many changes I went through :D I know it's related to the comment above so we may want to discuss this a bit more, but for now, I just made it non-throwing.